include/deal.II/base/vectorization.h

```diff
@@ -676,6 +677,31 @@ inline DEAL_II_ALWAYS_INLINE VectorizedArrayType
 
 
 
+/**
+ * Load @p n_array_elements from memory into the the VectorizedArray @p out,
+ * starting at the given addresses and with given offset, each entry from the
+ * offset providing one element of the vectorized array.
+ *
+ * This operation corresponds to the following code:
+ * @code
+ * for (unsigned int v=0; v<VectorizedArray<Number>::n_array_elements; ++v)
+ *   out.data[v] = ptrs[v][offset];
+ * @endcode
+ */
+template <typename Number, int width, std::size_t width_>
+inline DEAL_II_ALWAYS_INLINE void
+gather(VectorizedArray<Number, width> &    out,
+       const std::array<Number *, width_> &ptrs,
+       const unsigned int                  offset)
+{
+  static_assert(width == width_, "Length of input vector do not match!");
+
+  for (unsigned int v = 0; v < width_; v++)
+    out.data[v] = ptrs[v][offset];
+}
```

Comment:
@kronbichler I have moved it into a free function. How do you like it this way?

Reply:
Nice!